### ISSUE TITLE
numpy.linalg.cholesky accepts invalid covariance matrices

### ISSUE THREAD

**Issue by Phillip-M-Feldman** [numpy/numpy#15134]
When `numpy.linalg.cholesky` is given a non-symmetric matrix, it should raise an appropriate exception.  Note that the following ipython session produces a result, but shouldn't:

```
In [21]: x
Out[21]:
array([[1. , 0. , 0. ],
       [0.5, 1. , 0. ],
       [0. , 0. , 1. ]])

In [22]: linalg.cholesky(x)
Out[22]:
array([[1.       , 0.       , 0.       ],
       [0.5      , 0.8660254, 0.       ],
       [0.       , 0.       , 1.       ]])
```

**Comment by eric-wieser** [comment]
NumPy just call the lapack `*potrf` functions here I think

**Comment by Phillip-M-Feldman** [comment]
It appears that `numpy.linalg.cholesky` only examines entries on or below
the main diagonal (see below).  It is probably sufficient to update the
documentation to state (a) that the function only ignores entries above the
main diagonal, and (b) that it is incumbent on the user to insure that the
matrix is in fact symmetric.

In [1]: x= eye(3)

In [2]: x[0, 1]= 0.1

In [3]: linalg.cholesky(x)
Out[3]:
array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

**Comment by eric-wieser** [comment]
Updating the docs sounds good to me

**Comment by tylerjereddy** [comment]
Fixed by #15153
